pkg/reconciler/broker/broker.go

```diff
@@ -0,0 +1,480 @@
+		if err != nil {
+			// For some reason (maybe some cache invalidation thing), sometimes t.Exists returns that the topic
+			// doesn't exist but it actually does. When we try to create it again, it fails with an AlreadyExists
+			// reason. We check for that error here. If it happens, then return nil.
+			if _, ok := gstatus.FromError(err); !ok {
+				logger.Error("Failed from Pub/Sub client while creating topic", zap.Error(err))
+				b.Status.MarkTopicFailed("CreationFailed", "Topic creation failed: %w", err)
+				return err
+			}
+			logger.Error("Failed to create Pub/Sub topic", zap.Error(err))
+			b.Status.MarkTopicFailed("CreationFailed", "Topic creation failed: %w", err)
+			return err
+		}
+		logger.Info("Created PubSub topic", zap.String("name", topic.ID()))
+		r.Recorder.Eventf(b, corev1.EventTypeNormal, topicCreated, "Created PubSub topic %q", topic.ID())
+	}
+
+	b.Status.MarkTopicReady()
+	// TODO(grantr): this isn't actually persisted due to webhook issues.
+	b.Status.TopicID = topic.ID()
+
+	// Check if PullSub exists, and if not, create it.
+	subID := resources.GenerateDecouplingSubscriptionName(b)
+	sub := client.Subscription(subID)
+	subExists, err := sub.Exists(ctx)
+	if err != nil {
+		logger.Error("Failed to verify Pub/Sub subscription exists", zap.Error(err))
+		return err
+	}
+
+	if !subExists {
+		subConfig := gpubsub.SubscriptionConfig{
+			Topic: topic,
+			Labels: map[string]string{
+				"resource":     "brokers",
+				"broker_class": brokerv1beta1.BrokerClass,
+				"namespace":    b.Namespace,
+				"name":         b.Name,
+				//TODO add resource labels, but need to be sanitized: https://cloud.google.com/pubsub/docs/labels#requirements
+			},
+			//TODO(grantr): configure these settings?
+			// AckDeadline
+			// RetentionDuration
+		}
+		// Create a new subscription to the previous topic with the given name.
+		logger.Debug("Creating sub with cfg", zap.String("id", subID), zap.Any("cfg", subConfig))
+		sub, err = client.CreateSubscription(ctx, subID, subConfig)
+		if err != nil {
+			logger.Error("Failed to create subscription", zap.Error(err))
+			b.Status.MarkSubscriptionFailed("CreationFailed", "Subscription creation failed: %w", err)
+			return err
+		}
+		logger.Info("Created PubSub subscription", zap.String("name", sub.ID()))
+		r.Recorder.Eventf(b, corev1.EventTypeNormal, subCreated, "Created PubSub subscription %q", sub.ID())
+	}
+	//TODO update the subscription's config if needed.
+
+	b.Status.MarkSubscriptionReady()
+	// TODO(grantr): this isn't actually persisted due to webhook issues.
+	b.Status.SubscriptionID = sub.ID()
+
+	return nil
+}
+
+func (r *Reconciler) deleteDecouplingTopicAndSubscription(ctx context.Context, b *brokerv1beta1.Broker) error {
+	logger := logging.FromContext(ctx).Desugar()
+	logger.Debug("Deleting decoupling topic")
+
+	// get ProjectID from config or metadata
+	//TODO(grantr) support configuring project in broker config
```

Comment:
We probably don't want to support multi project for now. Project ID can be a property of the controller.

Reply:
Removed this TODO.